framework/src/problems/FEProblemBase.C

```diff
@@ -2386,6 +2387,29 @@ FEProblemBase::getDistribution(const std::string & name)
   return *(objs[0]);
 }
 
+void
+FEProblemBase::addLikelihood(const std::string & type,
+                             const std::string & name,
+                             InputParameters & parameters)
+{
+  parameters.set<std::string>("type") = type;
+  addObject<Likelihood>(type, name, parameters, /* threaded = */ false);
+}
+
```

Comment:
I'd create an action in STM and add the Likelihood like this:
```
_problem->addObject<MappingBase>(_type, _name, _moose_object_pars, /* threaded = */ false);
```
I think the warehouse query will take care of the cast.